### PR TITLE
Location filtering

### PR DESCRIPTION
**Pull request by martinamps**
Adds the ability to search an area:

```
➜  ~ curl -s -X GET 'localhost:5000/api/v1/location?source_latitude=30.267153&source_longitude=-97.743057&distance=5' | jq '.[0]'
{
  "additional_information_for_patients": "",
  "created_on": "Wed, 25 Mar 2020 21:32:00 GMT",
  "data_source": "[GISCorps] ",
  "deleted_on": null,
  "external_location_id": "[{\"use\": \"primary\", \"kind\": \"esriFieldTypeOID\", \"alias\": \"OBJECTID\", \"field\": \"OBJECTID\", \"value\": \"63\", \"system\": \"Esri\", \"assigner\": \"GISCorps\"}, null, {\"use\": \"other\", \"kind\": \"esriFieldTypeGlobalID\", \"alias\": \"GlobalID\", \"field\": \"GlobalID\", \"value\": \"08824bf4-e36f-48d7-961e-5f814a9d7471\", \"system\": \"Esri\", \"assigner\": \"Esri\"}]",
  "geojson": {},
  "is_collecting_samples": false,
  "is_collecting_samples_by_appointment_only": false,
  "is_collecting_samples_for_others": true,
  "is_collecting_samples_onsite": false,
  "is_evaluating_symptoms": false,
  "is_evaluating_symptoms_by_appointment_only": false,
  "is_hidden": false,
  "is_ordering_tests": false,
  "is_ordering_tests_only_for_those_who_meeting_criteria": true,
  "is_processing_samples": false,
  "is_processing_samples_for_others": false,
  "is_processing_samples_onsite": false,
  "is_verified": true,
  "location_address_locality": "",
  "location_address_postal_code": "",
  "location_address_region": "",
  "location_address_street": "1601 Trinity St. Austin, Tx 78701",
  "location_contact_phone_appointments": "",
  "location_contact_phone_covid": "",
  "location_contact_phone_main": "833-882-2737",
  "location_contact_url_covid_appointments": "",
  "location_contact_url_covid_info": "",
  "location_contact_url_covid_screening_tool": "",
  "location_contact_url_covid_virtual_visit": "",
  "location_contact_url_main": "https://dshs.texas.gov/coronavirus/testing.aspx",
  "location_hours_of_operation": "8AM-5PM (M-F)",
  "location_id": "c6a0fcbe-5f6b-c05c-be06-b0298d63874d",
  "location_latitude": 30.2768457397342,
  "location_longitude": -97.7355596703593,
  "location_name": "UT Health Austin (Dell Medical Clinical Practice)",
  "location_place_of_service_type": "Other",
  "location_specific_testing_criteria": "Hours of operation are 8AM-5PM (M-F) but are subject to change without notice. Appointments are required at this location. Onsite staff is able to offer COVID-19 testing only (no screening). Testing is only performed for individuals who meet testing criteria. This location was operating normally as of our last check-in. (Updated: 2020-04-20 | Source: https://dshs.texas.gov/coronavirus/testing.aspx)",
  "location_status": "Operational",
  "raw_data": "{}",
  "record_id": 1136,
  "reference_publisher_of_criteria": "",
  "updated_on": "Fri, 24 Apr 2020 12:50:49 GMT"
}
```

### CODE REVIEW

File: application/database/database_routes.py [hunk 1]
```diff
@@ -20,15 +20,20 @@ def s3_upload(file_path, bucket_name):
...
-  if s is None:
-    pass
-  else:
-    return s.lower() in ['true', 'TRUE', 'True', '1', 'yes', 'y', 't']
+  if type(s) == bool:
+    return s
+  elif type(s) == int:
+    return s == 1
+  elif type(s) == str:
+    return s.lower() in ['true', '1', 'yes', 'y', 't']
```
Comment (jorge-caballero) [inline application/database/database_routes.py:33]: Should the `1` still be in here? If not, prob best to take out

File: application/database/database_routes.py [hunk 2]
```diff
@@ -55,8 +60,21 @@ def get_recent_location():
...
   """
-  table_data = Entities.query.order_by(Entities.record_id).all()
+
+  filter_latitude = request.args.get('source_latitude')
+  filter_longitude = request.args.get('source_longitude')
+  filter_distance = request.args.get('distance') # miles
+
+  if filter_latitude and filter_longitude and filter_distance:
+    try:
+      table_data = Entities.query.filter(text("POINT({:f}, {:f}) <@> POINT(location_longitude, location_latitude) < {:f}".format(float(filter_longitude), float(filter_latitude), float(filter_distance)))).order_by(Entities.record_id)
```
Comment (jorge-caballero) [inline application/database/database_routes.py]: we're not listing these points on the front end...i recommend removing the sorting call in order to minimize performance overhead. this isn't a huge issue now, but will become an issue shortly as the number of records doubles and triples in the next week or so

File: application/database/database_routes.py [hunk 3]
```diff
@@ -55,8 +60,21 @@ def get_recent_location():
...
+    except:
+      abort(400)
+  else:
+    table_data = Entities.query.order_by(Entities.record_id).all()
```
Comment (jorge-caballero) [inline application/database/database_routes.py]: ditto re: sorting.

**Review by jorge-caballero — APPROVED** [review]

File: application/database/database_routes.py (hunk 2 above)
Comment (martinamps) [reply application/database/database_routes.py]: given it's the primary key it's all in memory so shouldn't actually make a material difference:

>    "entities_pkey" PRIMARY KEY, btree (record_id)
>    "ix_entities_location_id" btree (location_id)

if we get to a point where we need to optimize queries I'll be happy ;) sorting is redundant tho so I will kill it

Comment (martinamps) [reply application/database/database_routes.py]: done

**Review by iloveicedgreentea — APPROVED** [review]

File: application/database/database_routes.py (hunk 1 above)
Comment (martinamps) [reply application/database/database_routes.py:33]: yes because `'1'` is still a string, `1` is not

**Pull request merged**